source/Octopus.Client/IOctopusAsyncRepository.cs

```diff
@@ -9,60 +11,87 @@ namespace Octopus.Client
         IDeploymentProcessRepository DeploymentProcesses { get; }
         IDeploymentRepository Deployments { get; }
         IEnvironmentRepository Environments { get; }
-        IEventRepository Events { get; }
-        IFeaturesConfigurationRepository FeaturesConfiguration { get; }
         IFeedRepository Feeds { get; }
         IInterruptionRepository Interruptions { get; }
         ILibraryVariableSetRepository LibraryVariableSets { get; }
         ILifecyclesRepository Lifecycles { get; }
         IMachinePolicyRepository MachinePolicies { get; }
         IMachineRepository Machines { get; }
         IMachineRoleRepository MachineRoles { get; }
-        IMigrationRepository Migrations { get; }
-        IOctopusServerNodeRepository OctopusServerNodes { get; }
-        IPerformanceConfigurationRepository PerformanceConfiguration { get; }
         IProjectGroupRepository ProjectGroups { get; }
         IProjectRepository Projects { get; }
         IProjectTriggerRepository ProjectTriggers { get; }
         IProxyRepository Proxies { get; }
         IReleaseRepository Releases { get; }
         IRetentionPolicyRepository RetentionPolicies { get; }
         ISchedulerRepository Schedulers { get; }
-        IServerStatusRepository ServerStatus { get; }
-        ISpaceRepository Spaces { get; }
         ISubscriptionRepository Subscriptions { get; }
         ITagSetRepository TagSets { get; }
-        ITaskRepository Tasks { get; }
-        ITeamsRepository Teams { get; }
         ITenantRepository Tenants { get; }
         ITenantVariablesRepository TenantVariables { get; }
-        IUserRepository Users { get; }
-        IUserRolesRepository UserRoles { get; }
         IVariableSetRepository VariableSets { get; }
         IWorkerPoolRepository WorkerPools { get; }
         IWorkerRepository Workers { get; }
+        IEventRepository Events { get; }
+        ITaskRepository Tasks { get; }
+        ITeamsRepository Teams { get; }
         IScopedUserRoleRepository ScopedUserRoles { get; }
         IUserPermissionsRepository UserPermissions { get; }
+        IBackupRepository Backups { get; }
+        ICommunityActionTemplateRepository CommunityActionTemplates { get; }
+        IConfigurationRepository Configuration { get; }
+        IFeaturesConfigurationRepository FeaturesConfiguration { get; }
+        IMigrationRepository Migrations { get; }
+        IOctopusServerNodeRepository OctopusServerNodes { get; }
+        IPerformanceConfigurationRepository PerformanceConfiguration { get; }
+        IServerStatusRepository ServerStatus { get; }
+        ISpaceRepository Spaces { get; }
+        IUserRepository Users { get; }
+        IUserRolesRepository UserRoles { get; }
+        /// <summary>
+        /// The client over which the repository is run.
+        /// </summary>
+        IOctopusAsyncClient Client { get; }
+        SpaceContext SpaceContext { get; }
+        Task<IOctopusAsyncRepository> ForSpaceContext(string spaceId);
```

Comment:
BTW this method should just be named `ForSpace`... remove the word `Context` here